### DISCUSSION TITLE
Is it possible to have two Docs like layouts?

### DISCUSSION THREAD

**Discussion by binnyg** [thuliteio/doks#538] [Q&A] [answered]
First of all, thank you and appreciate the great theme and your contribution.

We have a master project which has two sub-projects. We would like to have two separate documentation sections on the main menu. Think of it as two Docs but named differently. How do I do that? Goal is to have the same layout as Docs for two main menu items.

**Comment by MattDodsonEnglish** [comment]
If I correctly understand what you want to do, I think you won't have much trouble. You can definitely have multiple sections that use the "docs" theme as the base design.

That's exactly what we did for our site, where we have "docs" and "data model": https://trueblocks.io/

Off the top of my head, there's 3 things you'll need to do:

1. Add a horizontal section in `menus.toml`. Add categories for the section's sidebar menu.
2. In layouts, either
   a) create a directory for your new section, with a new  `single.html` and `list.html` menus (copying heavily from the single and list pages in `layouts/docs`)
    b) get a little DRYer and use the single and list layouts in `layout/default` (might need to modify this)
 3. In `content`, add your new section's content.
 
 @h-enk  Himself did a demonstration of how to do this here. 
 - https://github.com/atwriter/new_doks_site/pull/1/commits
 
 You're also welcome to look at how we made a "Data model" section here (you might find some noobisms :smile:). If I recall we changed the structure of the two sections in some subtle ways, but I'm not sure.
 - https://github.com/TrueBlocks/trueblocks-docs
 
 Note that both of those examples are using an early version of Doks, the procedure may have changed slightly now.

**Reply by binnyg** [reply]
I was able to duplicate the docs layout, docs-navigation.html, and partials (docs-menu.html, and docs-toc.html) to achieve what I wanted. But I wonder if there is a smart way to reuse the layout without duplicating the template files, after all, it is the same exact layout we are trying to use for other sub-projects.

BTW, I couldn't make it work when two sections had the same file name. I had to make sure file names are unique across content folders.

Thank you Matt for willing to help.

**Reply by binnyg** [reply]
The duplicate file name issue is coming from the reference to the page and not the page itself. 

`Get help on Doks. [Help →]({{< relref "how-to-update" >}})`

I guess I will duplicate the layout for now until there is another way to do it.

**Reply by h-enk** [reply]
You would have make the reference more specific/reference it uniquely, i.e. add the section like so:

```md
Get help on Doks. [Help →]({{< relref "section-1/how-to-update" >}})
```

**Comment by h-enk** [comment]
> How do I do that? Goal is to have the same layout as Docs for two main menu items.

I'm aware that this currently is a hassle. It's on the list to automate this/making it a breeze.

**Reply by iamsantanubanerjee** [reply]
Thank you so much for working on it. When can we expect it to land? And will it be available through the docs-child-theme?

**Reply by h-enk** [reply]
It's already available — with [Doks v0.4.0](https://github.com/h-enk/doks/releases/tag/v0.4.0) (both starter and child theme).

However, the documentation still needs to be updated, I'm working on it — will be available soon

**Reply by mmasko** [reply]
Is the documentation complete for this? If I'm understanding correctly, this is so we can have two (or more) unique documentation sections, correct? I can't seem to find this in the documentation for the improvements made in [Doks v0.4.0](https://github.com/h-enk/doks/releases/tag/v0.4.0). The automated capabilities I mean. 

Thank you.

**Reply by h-enk** [reply, marked as answer]
Yes, but I guess it's hidden. I'm considering to set up the new documentation more from a use case perspective — hoping that's more helpful

Create a docs based tree — with a single command:

```bash
npm run create -- --kind docs [section]
```

For example, create a docs based tree named guides:

```
npm run create -- --kind docs guides
```

See also: [Docs based tree](https://getdoks.org/docs/overview/commands/#docs-based-tree)

Note that, you have to make sure to use unique identifiers in the frontmatter of the pages

**Reply by h-enk** [reply]
> If I'm understanding correctly, this is so we can have two (or more) unique documentation sections, correct? 

And yes, that's correct

**Reply by mmasko** [reply]
Ok, thank you. I think I'm understanding this now. It does look like there's some odd behavior where the command forces the new docs section to literally be a child of lorem/ipsum. Then all new pages are also forced to be within the folder structure. I'll open an issue in GH though so your team can properly analyze it. 

Thanks again!

**Reply by binnyg** [reply]
Thanks for adding this feature. I am trying to test the new "create kind docs" feature but I couldn't make the navigation work. I was able to successfully create a docs section but the sidebar navigation is missing. Do I have do anything special to make the navigation visible?

**Reply by h-enk** [reply]
In `./config/_default/params.toml` set the sections you'd like to show section navigation for:

```toml
[..]
[sections]
  sectionNav = ["docs", "guides"]
[..]
```
